clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1127,6 +1145,21 @@ static void populateIntHeader(SYCLIntegrationHeader &H, const StringRef Name,
       uint64_t Sz = Ctx.getTypeSizeInChars(Fld->getType()).getQuantity();
       H.addParamDesc(SYCLIntegrationHeader::kind_pointer,
                      static_cast<unsigned>(Sz), static_cast<unsigned>(Offset));
+    } else if (Util::isSyclSpecConstantType(ArgTy)) {
+      // Add specialization constant ID to the header.
+      auto *TmplSpec =
+          cast<ClassTemplateSpecializationDecl>(ArgTy->getAsCXXRecordDecl());
+      const TemplateArgumentList *TemplateArgs =
+          &TmplSpec->getTemplateInstantiationArgs();
+      constexpr unsigned int TypeIDTmplArgNo = 1;
```

Comment:
Whats the purpose of this?  Is it really only used the 1x below?  I think I'd rather it just be get(1)

Reply:
Also, do we just ignore the first argument?

Reply:
> Whats the purpose of this? Is it really only used the 1x below? I think I'd rather it just be get(1)

It is self-commenting var name for the index '1'. Yes, it is used once, but my idea is that it is
better use properly named constant rather then literal, similarly to `#define SOME_DESCRIPTIVE_NAME val`

> Also, do we just ignore the first argument?

Here in FE - yes. It is used in the runtime though.

@erichkeane - given the explanations, are you OK with not changing the code here?

Reply:
IMO it doesn't make it more readable. I'd still prefer it just be get(1). Add a comment that it is the typeid if you want (actually a good descriptive comment would be appreciated).

Reply:
ok